src/timepicker/timepicker.js

```diff
@@ -114,6 +120,39 @@ angular.module('ui.bootstrap.timepicker', [])
 
   };
 
+  // Respond on up/down arrowkeys
+  this.setupArrowkeyEvents = function( hoursInputEl, minutesInputEl ) {
+    hoursInputEl.bind('keydown', function(e) {
+      if ( e.which === 38 ) { // up
+        $scope.$apply(function() {
+          $scope.incrementHours();
```

Comment:
This should be rewritten as
<pre>
$scope.incrementHours();
$scope.$apply();
</pre>